api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
Well, I thought we can throw an error, or at least we need to leave a log that ```hasHeaderRow``` / ```skipHeaderRows``` would be ignored so that users can figure out what happened by reading their log.